crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -686,6 +686,24 @@ impl CumulocityConverter {
                     vec![]
                 }
             }
+            C8yDeviceControlOperation::DeviceProfile(request) => {
+                if self.config.capabilities.device_profile {
+                    if let Some(profile_name) = extras.get("profileName") {
```

Comment:
Though, `profileName` is not a part of `C8yDeviceProfile` but `C8yOperation`  and it looks like this:
```
"id": "123456",
"profileName": "test-profile",
"c8y_DeviceProfile": {
    /*...*/
},
"externalSource": {
    /*...*/
}
```
I could merge those two json values into one and pass to `C8yDeviceProfile` struct but not sure it is viable solution. It would look like this:
```rust
else if let (Some(c8y_value), Some(name_value)) =
            (hashmap.get("c8y_DeviceProfile"), hashmap.get("profileName"))
        {
            let c8y_value_with_name = {
                c8y_value
                    .clone()
                    .as_object_mut()
                    .unwrap()
                    .append(name_value.clone().as_object_mut().unwrap());
                ().into()
            };
            C8yDeviceControlOperation::DeviceProfile(C8yDeviceProfile::from_json_value(
                c8y_value_with_name,
            )?)
```

Reply:
> Though, profileName is not a part of C8yDeviceProfile but C8yOperation and it looks like this:

I don't get your comment. I thought `C8yDeviceProfile` is a representation of `c8y_DeviceProfile` operation. What I am saying is like this:

```rust
#[derive(Debug, Deserialize, Eq, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct C8yDeviceProfile {
    pub profile_name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub firmware: Option<C8yFirmware>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub software: Option<C8ySoftwareUpdate>,
    #[serde(default, skip_serializing_if = "Vec::is_empty")]
    pub configuration: Vec<C8yDownloadConfigFile>,
}
```

Does serde deserilization doesn't work with this?

Reply:
Exactly, I was also disappointed with current solution.